### DISCUSSION TITLE
Refactor Favorites functionality of Aliases on the Dashboard

### DISCUSSION THREAD

**Discussion by WhamboMPS** [simple-login/app#577] [Feature Requests]
An Alias card can be marked by the user as a "Favorite". Doing so will ensure that the card auto-sorts to the top of the list on the [Dashboard](https://app.simplelogin.io/dashboard/), a handy feature. However, this has unintended consequences.

When a custom alias is created this creation process occurs on a new web page, [New Custom Alias](https://app.simplelogin.io/dashboard/custom_alias). Clicking the "Create" button on the New Custom Alias page returns the user to the Dashboard (as would be expected). However, this new alias will not appear at the top of the Dashboard but will instead show up below any previously marked Favorite alias(es). This means the user has to scroll down to find it.

Instead of auto-sorting Favorites to the top of the Dashboard, include them as an item in the filter criterion that defaults to "All Aliases".

![SL Filter criterion](https://user-images.githubusercontent.com/88680678/129416046-7a3304f4-f096-4960-aa7e-4c9e778139d0.jpg)

**Comment by nguyenkims** [comment]
@WhamboMPS a new filter for only pinned aliases is now added 

![2021-09-18 at 19 22](https://user-images.githubusercontent.com/3167780/133897219-1bbf23e5-0ca3-4c02-bbf4-10ec3db5061c.png)

The display after an alias creation is also improved to make sure the new alias is correctly highlighted :).

**Reply by WhamboMPS** [reply]
@nguyenkims, oh I so hate to be be ungrateful when someone has done something specifically at my request but I think this isn't quite there yet. Yes, it is now easy to filter to just _Pinned Aliases_ (A.K.A. _Favorites_). However, aliases that are pinned continue to sort to the top of the list regardless of the filter box's selection. Now that you have made it easy to filter to those Alias(es) they should no longer sort to the top of the list; as implemented the enhancement does not actually address the issue in the original Feature Request. Sorry....

Request 1: No longer sort Favorited Alias(es) to the top of the list; instead use the sort specified in the Sort dropdown (e.g. _Sort by most recent activity_). If a user wants to see their favorite aliases they should use the _Pinned Aliases_ feature to expose them.

Request 2: As a recommendation choose to name the underlying capability a _Favorite_ **or** a _Pinned Alias_. Under the current implementation the alias card has a Call To Action ("CTA") labeled _Add to favorite_ while the filter box is labeled _Pinned Aliases_. To avoid confusion you should choose only one name for this capability.

**Reply by nguyenkims** [reply]
@WhamboMPS you're right that the wording is a bit confusing, I've updated to use "pin" instead of "favorite" to avoid the confusion :)

> However, aliases that are pinned continue to sort to the top of the list regardless of the filter box's selection

I just tested and if a filter is selected, the pinned aliases will be ignored. The pinned aliases will always be on top as they are meant to be quickly accessed but they should disappear whenever an explicit filter is applied.

**Reply by WhamboMPS** [reply]
Thanks for looking into this, @nguyenkims. Unfortunately, I am not seeing the same behavior as you. [Link previously in this post removed 2021-09-21].

**Reply by nguyenkims** [reply]
@WhamboMPS this is indeed a bug, it has been fixed now in the latest version. Can you please retry?

**Reply by WhamboMPS** [reply]
@nguyenkims Just tested and can confirm that the feature is working as (I) would expect. Thanks for implementing this request!

**Reply by nguyenkims** [reply]
Cool!
